samples/javascript_nodejs/57.teams-conversation-bot/bots/teamsConversationBot.js

```diff
@@ -180,39 +174,42 @@ class TeamsConversationBot extends TeamsActivityHandler {
     async messageAllMembersAsync(context) {
         const members = await this.getPagedMembers(context);
 
-        members.forEach(async (teamMember) => {
+        await Promise.all(members.map(async (member) => {
             const message = MessageFactory.text(
-                `Hello ${teamMember.givenName} ${teamMember.surname}. I'm a Teams conversation bot.`
+                `Hello ${member.givenName} ${member.surname}. I'm a Teams conversation bot.`
             );
 
-            var ref = TurnContext.getConversationReference(context.activity);
-            ref.user = teamMember;
+            const ref = TurnContext.getConversationReference(context.activity);
+            ref.user = member;
 
-            await context.adapter.createConversation(ref, async (t1) => {
-                const ref2 = TurnContext.getConversationReference(t1.activity);
-                await t1.adapter.continueConversation(ref2, async (t2) => {
-                    await t2.sendActivity(message);
+            await context.adapter.createConversation(ref, async (context) => {
```

Comment:
Modern Javascript supports block-level re-declaration of names. No need for disambiguation names, especially when they don't add to the readability of the code.